File: problems/238-product-of-array-except-self/python/combined_left_right.py
from typing import List


class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        result: List[int] = [0 for _ in nums]

        for i in range(len(nums)):
            left: int = 1
            right: int = 1

            for j in range(len(nums)):
                n: int = nums[j]
                if j < i:
                    left *= n
                elif j > i:
                    right *= n

            result[i] = left * right

        return result


def test(nums: List[int]) -> None:
    result: int = Solution().productExceptSelf(nums)
    print(f"Product except self for these numbers is {
          result} \n -> numbers: {nums}")


if __name__ == "__main__":
    test([1, 2, 3, 4])
    test([-1, 1, 0, -3, 3])

File: problems/238-product-of-array-except-self/csharp/combined_left_right.cs
using System.Linq;
using System;

public static class Program
{
  public static void Main(string[] args)
  {
    Solution s = new Solution();
    Console.WriteLine(string.Join(", ", s.ProductExceptSelf(new int[] { 1, 2, 3, 4 })));
    Console.WriteLine(string.Join(", ", s.ProductExceptSelf(new int[] { -1, 1, 0, -3, 3 })));
  }
}

public class Solution
{
  public int[] ProductExceptSelf(int[] nums)
  {
    int[] result = new int[nums.Length];

    for (int i = 0; i < nums.Length; i++)
    {
      int left = 1;
      int right = 1;

      for (int j = 0; j < nums.Length; j++)
        if (j < i)
          left *= nums[j];
        else if (j > i)
          right *= nums[j];

      result[i] = left * right;
    }

    return result;
  }
}